contracts/strategies/pangolin/strategy-png-usdc-ust.sol

```diff
@@ -0,0 +1,171 @@
+        address _timelock
+    )
+        public
+        StrategyPngMiniChefFarmBase(
+            _poolId,
+            png_usdc_ust_lp,
+            _governance,
+            _strategist,
+            _controller,
+            _timelock
+        )
+    {}
+
+    function _takeFeeUsdcToSnob(uint256 _keep) internal {
+        address[] memory path = new address[](3);
+        path[0] = usdc;
+        path[1] = wavax;
+        path[2] = snob;
+        IERC20(ust).safeApprove(pangolinRouter, 0);
+        IERC20(ust).safeApprove(pangolinRouter, _keep);
+        _swapPangolinWithPath(path, _keep);
+        uint256 _snob = IERC20(snob).balanceOf(address(this));
+        uint256 _share = _snob.mul(revenueShare).div(revenueShareMax);
+        IERC20(snob).safeTransfer(
+            feeDistributor,
+            _share
+        );
+        IERC20(snob).safeTransfer(
+            IController(controller).treasury(),
+            _snob.sub(_share)
+        );
+    }
+
+    function _takeFeeUstToSnob(uint256 _keep) internal {
+        address[] memory path = new address[](3);
+        path[0] = ust;
+        path[1] = wavax;
+        path[2] = snob;
+        IERC20(ust).safeApprove(pangolinRouter, 0);
+        IERC20(ust).safeApprove(pangolinRouter, _keep);
+        _swapPangolinWithPath(path, _keep);
+        uint256 _snob = IERC20(snob).balanceOf(address(this));
+        uint256 _share = _snob.mul(revenueShare).div(revenueShareMax);
+        IERC20(snob).safeTransfer(
+            feeDistributor,
+            _share
+        );
+        IERC20(snob).safeTransfer(
+            IController(controller).treasury(),
+            _snob.sub(_share)
+        );
+    }
+
+    // **** State Mutations ****
+
+    function harvest() public override onlyBenevolent {
+        // Collects Token Fees
+        IMiniChef(miniChef).harvest(poolId, address(this));
+
+        uint256 _png = IERC20(png).balanceOf(address(this));
+        uint256 _usdc = IERC20(usdc).balanceOf(address(this));
+        uint256 _ust = IERC20(ust).balanceOf(address(this));
+        
+        if (_png > 0) {
+            // 10% is sent to treasury
+            uint256 _keep = _png.mul(keep).div(keepMax);
+            if (_keep > 0) {
+                _takeFeePngToSnob(_keep);
+            }
+
+            _png = IERC20(png).balanceOf(address(this));  
+        }
+
+        if (_usdc > 0) {
+            uint256 _keep2 = _usdc.mul(keep).div(keepMax);
+            if (_keep2 > 0){
+                _takeFeeUsdcToSnob(_keep2);
+            }
+            
+            _usdc = IERC20(usdc).balanceOf(address(this));
+        }
+
+        if (_ust > 0) {
+            uint256 _keep2 = _ust.mul(keep).div(keepMax);
+            if (_keep2 > 0){
+                _takeFeeUstToSnob(_keep2);
+            }
+            
+            _ust = IERC20(ust).balanceOf(address(this));
+        }
+
+        // In the case of PNG Rewards, swap PNG for USDC and UST
+        _png = IERC20(png).balanceOf(address(this));
```

Comment:
I've already checked the `_png` balanceof on line 87...unsure if this should be removed, or if it costs little enough to leave in.

Reply:
This is true for wavax and luna as well.  Each time a fee is deducted, the _luna, _png, _wavax variables are updated accordingly.  So it is not needed, but it does make each codeblock easy to read.